### DISCUSSION TITLE
Sending document template message with an action

### DISCUSSION THREAD

**Discussion by Rodfor** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#49] [Q&A] [answered]
Hello

I was wondering it its possible to send a template message that has both a document in the header and a button/action (in this case a dynamic url)? For previous templates with an action I used the MPMParameter but this doesn't have a File field. 
(And the DocumentMessageParameter doesn't have a SubType/index field).

**Comment by gabrieldwight** [comment]
Hi, it is possible if the template manager allows to add both document file and buttons in the header supported when creating the template message. Currently I only know the template message header only support one type of data specified either text or media.

**Reply by Rodfor** [reply]
Hi, you are correct that it isn't possible to include both a document and a button in the header of a template but I am talking about 2 seperate components in 1 message : 1) a header component containing only a document and 2) a 'button component' which contains the dynamic url parameters. For 1) the only component object that seems to fit is a DocumentMessageComponent that can contain a DocumentMessageParameter which can include a Document object and for component 2) I'm using a MPMComponent which has the SubType and Index fields. But there seems to be no way to combine these 2 components in 1 message object. The closest example in the Cloud Api documentation I could find was [https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-message-templates#interactive](url) which uses an Image instead of  a document.
I hope this clarifies my question.

**Reply by gabrieldwight** [reply, marked as answer]
To combine both document and button into a message object you will need to create a custom interactive message, the header can contain the document media component and it will have a button component that can use the dynamic url parameters that will basically redirect to your endpoint from whatsapp when clicked since the call to action has limited functions you can perform with it.
